drivechain-server/Justfile

```diff
@@ -1,14 +1,41 @@
-build:
+build-go:
     go build -o drivechain-server .
 
-lint: 
+build: build-bdk-cli build-go
+
+clean:
+    rm -rf ./bdk/bin
+    go clean
+
+lint:
     golangci-lint run ./...
 
-gen: 
+gen:
     buf generate
 
 install-bdk-cli:
     cargo install \
         --git https://github.com/bitcoindevkit/bdk-cli \
         --no-default-features \
-        --features=key-value-db,compiler,electrum  
\ No newline at end of file
+        --features=key-value-db,compiler,electrum
+
+build-bdk-cli:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    if [ ! -f "./bdk/bin/bdk-cli" ]; then
+        echo "Building bdk-cli..."
+        mkdir -p ./bin
+        if ! command -v rustc &> /dev/null; then
+            echo "Error: Rust is not installed. Please install Rust and try again."
+            exit 1
+        fi
+        cargo install \
+        --git https://github.com/bitcoindevkit/bdk-cli.git \
+        --tag v0.27.1 \
+        --root ./bdk \
+        --no-default-features \
+        --features=key-value-db,compiler,electrum
```

Comment:
Could we extract these features out to a variable, and reuse in `install-bdk-cli`?

Reply:
Probably, let me just read the Justfile docs!

Reply:
Done